packages/react-native/src/handlers/UnhandledExceptionHandler.ts

```diff
@@ -0,0 +1,113 @@
+            }
+            client.send(error, {
+                'error.type': 'Unhandled exception',
+                fatal,
+            });
+            globalErrorHandler(error, fatal);
+        });
+    }
+
+    public captureUnhandledPromiseRejections(client: BacktraceClient) {
+        const hermesInternal = hermes();
+
+        if (hermesInternal?.hasPromise?.() && hermesInternal?.enablePromiseRejectionTracker) {
+            hermesInternal.enablePromiseRejectionTracker({
+                allRejections: true,
+                onUnhandled: (id: number, rejection: Error | object = {}) => {
+                    if (!this.enabled) {
+                        return;
+                    }
+                    client.send(
+                        new BacktraceReport(
+                            rejection as Error,
+                            {
+                                'error.type': 'Unhandled exception',
+                                unhandledPromiseRejectionId: id,
+                            },
+                            [],
+                            {
+                                classifiers: ['UnhandledPromiseRejection'],
+                                skipFrames: rejection instanceof Error ? 0 : 1,
+                            },
+                        ),
+                    );
+                },
+            });
+        } else {
+            rejectionTracking.enable({
+                allRejections: true,
+                onUnhandled: (id: number, rejection: Error) => {
+                    if (this.enabled) {
+                        client.send(
+                            new BacktraceReport(
+                                rejection as Error,
+                                {
+                                    'error.type': 'Unhandled exception',
+                                    unhandledPromiseRejectionId: id,
+                                },
+                                [],
+                                {
+                                    classifiers: ['UnhandledPromiseRejection'],
+                                    skipFrames: rejection instanceof Error ? 0 : 1,
+                                },
+                            ),
+                        );
+                    }
+                    if (!__DEV__) {
+                        return;
+                    }
+                    let message: string;
+                    let stack: string | undefined;
+
+                    const stringValue = Object.prototype.toString.call(rejection);
+                    if (stringValue === '[object Error]') {
+                        message = Error.prototype.toString.call(rejection);
+                        stack = rejection.stack as string;
+                    } else {
+                        try {
+                            // eslint-disable-next-line @typescript-eslint/no-var-requires
+                            message = require('pretty-format')(rejection);
```

Comment:
Can we require it once?

For example:
```
function cachedPrettyFormat() {
  let prettyFormat: any | false;
  return function cachedPrettyFormat() {
    if (prettyFormat === false) {
      return undefined;
    }

    if (prettyFormat !== undefined) {
      return prettyFormat;
    }

    try {
      return (prettyFormat = require('pretty-format'));
    } catch {
      prettyFormat = false;
      return undefined;
    }    
  }
}

const getPrettyFormat = cachedPrettyFormat();

// then

const prettyFormat = getPrettyFormat();
if (prettyFormat) {
  message = prettyFormat(rejection);
} else {
  message = typeof rejection === 'string' ? rejection : JSON.stringify({ ...rejection });
}
```

Reply:
The code you see here is a copy-paste code from the react-native repository to just do literally all steps the same way they did to easily follow up what they did and what are we doing. Reference:

https://github.com/facebook/react-native/blob/a59b947a1e077d1c3f0d36926d374db5fe7d3291/packages/react-native/Libraries/promiseRejectionTrackingOptions.js#L38
Anyway I believe its useful to have this change so:
- I will add your change 
- I will add a reference to the code I'm talking about